### ISSUE TITLE
Linking Against ERDAS ECW JP2 SDK 5.5 with QGIS Server

### ISSUE THREAD

**Issue by clifftanhy** [qgis/QGIS#55327]
### What is the issue about the documentation?

I have recently purchased the license for ERDAS ECW JP2 SDK 5.5. I am trying to serve ECW layers through a QGIS WINDOWSS server. I am not sure how can I do that? I do not know how to link the SDK license to QGIS? GDAL? MAPSERVE?

**Comment by NyakudyaA** [comment]
You basically need to compile Gdal to read the ECW. Once that is setup your Desktop or Server will be able to read them.
https://trac.osgeo.org/gdal/wiki/ECW

**Comment by clifftanhy** [comment]
Thanks mate, I have looked into that link there, however I am unable to locate **nmake.opt** in the file directory. is there something that I did not select while installing the QGIS server?

**Comment by jef-n** [comment]
I don't think you need to rebuild GDAL.  GDAL in OSGeo4W already has ECW support.  But you'll have to rebuild QGIS w/o `SERVER_SKIP_ECW=FALSE`, which is default for packaged builds in OSGeo4W.  Build recipes for the OSGeo4W packages are at [jef-n/OSGeo4W](https://github.com/jef-n/OSGeo4W/).

https://github.com/jef-n/OSGeo4W/blob/master/src/qgis/osgeo4w/package.sh#L144
https://github.com/jef-n/OSGeo4W/blob/master/src/qgis-ltr/osgeo4w/package.sh#L130

**Comment by clifftanhy** [comment]
I am very new to this; may I know how would one utilize the build recipes on a window server?

**Comment by rouault** [comment]
> I don't think you need to rebuild GDAL. GDAL in OSGeo4W already has ECW support. But you'll have to rebuild QGIS w/o `SERVER_SKIP_ECW=FALSE`

A build time setting for that seems extreme, doesn't it ? Perhaps QGIS server could skip the ECW driver by default, unless the user explicitly sets a QGIS_SERVER_ENABLE_ECW_BECAUSE_I_HAVE_A_ECW_LICENSE=YES environment variable. That way the default install would be safe from a legal point of view, and users would clearly take their responsibility if enabling the option.

**Comment by jef-n** [comment]
> A build time setting for that seems extreme, doesn't it ? Perhaps QGIS server could skip the ECW driver by default, unless the user explicitly sets a QGIS_SERVER_ENABLE_ECW_BECAUSE_I_HAVE_A_ECW_LICENSE=YES environment variable. That way the default install would be safe from a legal point of view, and users would clearly take their responsibility if enabling the option.

IANAL. It has been that way since 4d3d678080e6732e90777b5acec001702b3f8228 - IIRC because there were doubts that shipping ECW enabled server software is ok.   See also [OSGeo4W #337](https://trac.osgeo.org/osgeo4w/ticket/337) and #15975 (which used to be [#6838](https://issues.qgis.org/issues/6838) mentioned there).  That compile time option was added to bring ECW support back.

**Comment by cmosig** [comment]
Has anyone had any luck obtaining the 5.5 ECW SDK? Their website does not appear to be working.

**Comment by rouault** [comment]
> Their website does not appear to be working.

seems you need to have an account now (didn't remember this was the case before), but trying to do so just go that their website is under maintenance... The joys of relying on closed source software...   If the 3.3 SDK (that had a kind of "semi open-source" license) is good enough for you, check https://github.com/rouault/libecwj2-3.3-builds

**Comment by cmosig** [comment]
Wow, thanks! I was able to compile gdal with that 3.3 SDK you shared and it can now read ECW. Just what I needed, amazing, you saved my day.
